src/xdocs/checks/coding/missingnullcaseinswitch.xml

```diff
@@ -0,0 +1,148 @@
+      <subsection name="Description" id="Description">
+        <p>
+          Checks that switch statement or expression that uses reference type in case labels has
+          a <code>null</code> case label.
+        </p>
+        <p>
+         Rationale: switch statements and expressions in Java throw a
+          <code>NullPointerException</code> if the selector expression evaluates
+          to <code>null</code>. With the introduction of patterns in case label,
+          It is now possible to integrate a null check within the switch,
+          eliminating the risk of <code>NullPointerException</code> and simplifies the code
+          as there is no need for an external null check before entering the switch.
+        </p>
+        <p>
+          Limitations due to not being type-aware:
+        </p>
+        <ul>
+          <li>
+             This check does not differentiate between primitive types and
+             their boxed counterparts in case labels.
+             Therefore, any case label with an integer will not violate this check,
+             even if it is a reference type <code>Integer</code>.
+          </li>
+          <li>
+            This check will not violate case labels with idents even if
+            this is an identifier of Enums. We can't tell if the identifier
+            is an Enum or just a normal final variable.
+          </li>
+        </ul>
+        <p>
+          See the
+          <a href="https://docs.oracle.com/javase/specs/jls/se22/html/jls-15.html#jls-15.28">
+          Java Language Specification</a> for more information about switch statements
+          and expressions.
+        </p>
+      </subsection>
+
+      <subsection name="Examples" id="Examples">
+        <p id="Example1-config">
+          To configure the check:
+        </p>
+        <source>
+&lt;module name=&quot;Checker&quot;&gt;
+  &lt;module name=&quot;TreeWalker&quot;&gt;
+    &lt;module name=&quot;MissingNullCaseInSwitch&quot;/&gt;
+  &lt;/module&gt;
+&lt;/module&gt;
+        </source>
+        <p id="Example1-code">
+          Example of violation:
+        </p>
+        <source>
+public class Example1 {
+  void testString(String obj) {
+    // violation below, 'switch using reference types should have a null case.'
+    switch (obj) {
+      case &quot;something&quot; : {}
+    }
+    switch (obj) {
+      case null : {}
+      case &quot;something&quot; : {}
+      default : {}
+    }
+  }
+  void testPatterns(Object obj) {
+    // violation below, 'switch using reference types should have a null case.'
+    switch (obj) {
+      case Integer i : {} break;
+      default : {}
+    }
+    switch (obj) {
+      case null : {} break;
+      case Integer i : {}
+      default : {}
+    }
+  }
+  void testPrimitives(int obj) {
+    switch (obj) {   // ok, this is a primitive type can't be null
+      case 1 : {}
+    }
+  }
+
+  void testBoxedType(Integer obj) {
+    // ok, we can't tell if '1' is of a reference or primitive type
```

Comment:
from another class outside this file or do you mean so inner class?

Reply:
Some class that exists in core Java

Reply:
what do yo think about this?
```Java
public class Example1 {

  void testString(String obj) {
    // violation below, 'switch using reference types should have a null case.'
    switch (obj) {
      case "something" : {}
    }
    switch (obj) {
      case null : {}
      case "something" : {}
      default: {}
    }
  }

  void testPatterns(Object obj) {
    // violation below, 'switch using reference types should have a null case.'
    switch (obj) {
      case Integer i : {} break;
      default : {}
    }
  }

  void testPrimitives(int obj) {
    switch (obj) {   // ok, this is a primitive type can't be null
      case 1 : {}
    }
  }

  void testBoxedType() {
    switch (getInteger()) { // ok, we can't tell if '1' is an Integer type
      case 1 : {}
    }
  }

  void testIdent() {
    // ok, we can't tell if Ident 'ONE' is of a reference type,
    // i.e. Enum or primitive type i.e. local f
    switch (getEnum()) {
      case E.ONE: {}
    }
  }

  enum E { ONE }
  public static final int ONE = 5;
  public static Integer getInteger() { return 1; }
  public static E getEnum() { return E.ONE; }
}
```

Reply:
please review the example again if it is good or still needs improvements I added the methods in the same class

Reply:
We should not be using the term "Ident" anywhere, unless we are referring to the actual TokenType, in which case we should make that clear. However, we should not be referencing TokenTypes at all in this example, since this is an implementation detail.

In both of your examples above, we _could_ tell the type of the given variable if we wanted to, since everything is in the example. We need to use methods/constants from core java to drive the point home that Checkstyle has no idea about the type of the value resulting from these expressions:

```
    // This example is ok, because Checkstyle is not type-aware. The
    // value of 't' could be an enum constant, a primitive, or a reference
    // type.
    void m1() {
        var t = TimeUnit.DAYS;
        switch (t) {
            ...
        }
    }
    
    // This is example is also ok, because we do not know the return type
    // of a given method.
    void m2() {
        var t = List.of().size();
        switch (t) {
            ...
        }
    }
```

Reply:
> In both of your examples above, we _could_ tell the type of the given variable if we wanted to, since everything is in the example. We need to use methods/constants from core java to drive the point home that Checkstyle has no idea about the type of the value resulting from these expressions:

Yes, understood thank you